qemu/tests/virtio_fs_multi_users_access.py

```diff
@@ -0,0 +1,81 @@
+from virttest import error_context, utils_test, utils_disk, utils_misc
+
+from provider import win_driver_installer_test, virtio_fs_utils
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Different users access the shared directory.
+
+    1) Create a shared directory for testing on the host.
+    2) Run the virtiofsd daemon on the host.
+    3) Boot a guest on the host.
+    4) Log into guest and create the users.
+    5) Mount the file system( If the guest is linux ).
+    6) Create a new file in the shared dir.
+    7) Get the md5 value of the file in the shared dir.
+    8) Logout user and login another user( if available ),
+       then repeat step5 to step6.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    add_user_cmd = params.get("add_user_cmd")
+    del_user_cmd = params.get("del_user_cmd")
+    driver_name = params.get("driver_name")
+    fs_target = params.get("fs_target")
+    fs_dest = params.get("fs_dest")
+    os_type = params.get("os_type")
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+
+    username = params.objects("new_user")
+    pwd = params.objects("new_pwd")
+
+    try:
+        error_context.context("Create the user(s) on guest...", test.log.info)
+        for idx in range(len(username)):
+            if os_type == "windows":
+                status = session.cmd_status(add_user_cmd % (username[idx],
+                                                            pwd[idx]))
+            else:
+                status = session.cmd_status(add_user_cmd % (pwd[idx],
+                                                            username[idx]))
+            if status != 0:
+                test.fail("Failed to create user!")
+
+        if os_type == "windows":
+            session = utils_test.qemu.\
+                windrv_check_running_verifier(session, vm, test, driver_name)
+            win_driver_installer_test.run_viofs_service(test, params, session)
```

Comment:
Do you plan to change this part to virtio_fs_utils.py ?

Reply:
> Do you plan to change this part to virtio_fs_utils.py ?

Hello @xiagao , from my side, I would say NO since this test environment should be prepared before basic_io_test. 
If you have any opinions, just let me know! Thanks!